app/core/src/main/java/com/fsck/k9/mailstore/K9BackendFolder.kt

```diff
@@ -151,14 +151,14 @@ class K9BackendFolder(
                 val forwarded = cursor.getInt(4) == 1
                 val flagsColumnValue = cursor.getString(5)
 
-                val flags = flagsColumnValue.split(',').map { Flag.valueOf(it) }.toMutableSet().apply {
+                val flags = if (flagsColumnValue.isNullOrBlank()) emptySet<Flag>()
```

Comment:
The code to serialize and deserialize Flag values should probably live in a separate class and we should add tests. Your change looks like it will return an empty list if the database field is empty even when one of `deleted`, `read`, `flagged`, `answered`, or `forwarded` is set.

Reply:
Oops, thanks, yes. There are brackets missing in front of `if` and `.toMutableSet()`